app/src/main/java/com/jetbrains/handson/mpp/mobile/StationsSpinnerAdapter.kt

```diff
@@ -0,0 +1,56 @@
+package com.jetbrains.handson.mpp.mobile
+
+import android.content.Context
+import android.graphics.Color
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.TextView
+
+
+class StationsSpinnerAdapter(
+    private val context: Context, textViewResourceId: Int,
+    private val stations: Array<Station>
+) : ArrayAdapter<Station>(context, textViewResourceId, stations) {
+
+    override fun getCount(): Int {
+        return stations.size
+    }
+
+    override fun getItem(position: Int): Station {
+        return stations[position]
+    }
+
+    override fun getItemId(position: Int): Long {
+        return position.toLong()
+        // TODO: position can be nullable?
+    }
+
+
+    // And the "magic" goes here
+    // This is for the "passive" state of the spinner
+    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+        // I created a dynamic TextView here, but you can reference your own  custom layout for each spinner item
+        val label = super.getView(position, convertView, parent) as TextView
+        label.setTextColor(Color.BLACK)
+        // Then you can get the current item using the values array (Users array) and the current position
+        // You can NOW reference each method you has created in your bean object (User class)
+        label.text = stations[position].name
+
+        // And finally return your dynamic (or custom) view for each spinner item
+        return label
+    }
+
+    // And here is when the "chooser" is popped up
+    // Normally is the same view, but you can customize it if you want
+    override fun getDropDownView(
+        position: Int, convertView: View?,
+        parent: ViewGroup
+    ): View {
+        val label = super.getDropDownView(position, convertView, parent) as TextView
+        label.setTextColor(Color.BLACK)
+        label.text = stations[position].name
+
+        return label
```

Comment:
Similar to the above, you can create and return the label with `.apply` which just looks nicer 😊 

```
    return label.apply {
        setTextColor(Color.BLACK)
        text = stations[position].name
    }
```